### DISCUSSION TITLE
Which actions and or workflows to implement?

### DISCUSSION THREAD

**Discussion by gbowne1** [gbowne1/reactsocialnetwork#159] [General]
GitHub allows both actions and workflows.. there are many pre-configured workflows and actions we could potentially use.

My ideas is to use the following

- .github/workflows/lint.yml
runs ESLint on build as pre-deploy function

- .github/workflows/labels.yml
creates labels for issues

- .github/workflows/ci.yml 
runs our integration testing

- .github/workflows/build.yml
runs the react build process

- .github/workflows/deploy.yml
deploys build folder to the physical server

- .github/workflows/dependabot.yml
keeps track of our npm/node dependencies

These are the exact filenames I would use and you can use github advanced search to find examples of each to build from.  Most of these could be run against macos, windows or linux.

I will entertain ideas @manuel12 @pawel975

**Comment by pawel975** [comment]
I should dive into this topic since I don't know anything about it.

**Reply by gbowne1** [reply]
Actioins and workflows are very useful.  It could help us immensely.

**Reply by manuel12** [reply]
> I should dive into this topic since I don't know anything about it.

Will do too, haven't read much about github actions/workflows.

**Reply by gbowne1** [reply]
They are pretty powerful.  I have some experience with watching the automation run.

**Reply by gbowne1** [reply]
the easy part is that it is just YML

**Comment by gbowne1** [comment]
@balajik
